tests/js/object-tests.js

```diff
@@ -724,4 +724,36 @@ module.exports = {
       }, 2000);
     });
   },
+
+  testGetPropertyType: function () {
+    const realm = new Realm({
+      schema: [schemas.AllTypes, schemas.TestObject, schemas.LinkToAllTypes],
```

Comment:
I would love to see a test using a property of type `mixed` as well. To verify that it's the underlying type and not simply the type from the schema which gets returned.